### ISSUE TITLE
SCSS + erroneous CSS in plugin

### ISSUE THREAD

**Issue by mahotilo** [Typesetter/Typesetter#210]
I ran into a problem. 
1. I have SCSS Theme. If I made an error in style.scss,  I have a completely broken theme styles. But I have also a yellow message with description: "SCSS Compile Failed: parse error at: ... on line ...".
2. I have style.css file in my plugin. If I made an error in style.css, I have a completely broken theme styles. But I have no message about my error and its location.

Given caching, all this does not happen right away and it's hard to immediately understand what has broken the site.

Can we localize the source of an error in this case also?

**Comment by juek** [comment]
Scss errors will stop the processing of the current set of passed scss files, which are for the theme/layout the variables.scss, style.scss and possible other imports. 

> If I made an error in style.css, I have a completely broken theme styles. 

For any plugin *.css files, they should not impact the theme at all, since they will only be loaded separately or, if _combine CSS files_ turned on, combined with processed scss/less and other css files. The only thing I can imagine causing one css file to impact others is if A) _combine CSS files_ is on and B) there is an open comment /* somewhere in the plugin css, which could turn everything following into a comment unless a termintor */ is found in another css.

**Comment by mahotilo** [comment]
It is .css from custom section type. Error is doubled {

**Comment by juek** [comment]
> Error is doubled {

Ah ok, double curly braces will cause similar problems as open comments. It will leave a declaration block unterminated, meaning that all the follwing css is part of the particular rule and no other element rules will appply unless the open block is terminated.

**Comment by mahotilo** [comment]
Yes. SCSS compiler detect such errors in scss files, but not in css. 
I can rename .css to .scss, and this give me error messages. But it is not suitable in all cases.

**Comment by thomasfrobieter** [comment]
Just my two cents about this ... every web developer has to deal with CSS bugs, your editor should show you syntax errors. So you should simply copy your CSS code into your editor and check for errors.

**Comment by juek** [comment]
> The problem is to detect where an error appear.

There are only a few cases where css errors may impact more than a single rule. Actually I cannot imagine more than the 2 we already mentioned. Pushing everything through the Scss or LESS processor may help to find errors but it's overhead. IMO it's not the job of the CMS to parse vanilla CSS for errors.

**Comment by mahotilo** [comment]
@thomasfrobieter Yes. Maybe it is question of editor select. My favourit Notepad2-mod sometimes confused with some comments and css3, so I often ignore its colour mark-up.
But if we will send all css to sass or less compiler, will we have a problems? If not, we will have error localizations.

**Comment by mahotilo** [comment]
> Pushing everything through the Scss or LESS processor may help to find errors but it's overhead. IMO it's not the job of the CMS to parse vanilla CSS for errors.

@juek Oh. I wrote abut right now. Ok, I wanted to know you opinion about. Thanks to all.

P.S.
I wrote about due to similarity of the aftermath of error in *.css and TS error during sections editing.  I experienced yesterday a dozens unpleasant minutes trying to restore the page after "crash" until I realized that the error was in the css of my plugin.
